spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -514,11 +515,11 @@ private Message<?> checkHeaders(Object result, String topic, @Nullable Object so
 			if (needsTopic) {
 				builder.setHeader(KafkaHeaders.TOPIC, topic);
 			}
-			if (needsCorrelation && sourceIsMessage) {
+			if (needsCorrelation) {
 				builder.setHeader(this.correlationHeaderName,
 						((Message<?>) source).getHeaders().get(this.correlationHeaderName));
 			}
-			if (sourceIsMessage && reply.getHeaders().get(KafkaHeaders.REPLY_PARTITION) == null) {
```

Comment:
confusing. should `REPLY_PARTITION` be relevant on `source` side? On `reply` side, seems only `PARTITION` header matters.